crates/sui-types/src/execution.rs

```diff
@@ -101,6 +99,72 @@ pub struct ExecutionResultsV2 {
     pub user_events: Vec<Event>,
 }
 
+impl ExecutionResultsV2 {
+    pub fn drop_writes(&mut self) {
+        self.written_objects.clear();
+        self.modified_objects.clear();
+        self.created_object_ids.clear();
+        self.deleted_object_ids.clear();
+        self.user_events.clear();
+    }
+
+    pub fn merge_results(&mut self, new_results: Self) {
+        self.written_objects.extend(new_results.written_objects);
+        self.modified_objects.extend(new_results.modified_objects);
+        self.created_object_ids
+            .extend(new_results.created_object_ids);
+        self.deleted_object_ids
+            .extend(new_results.deleted_object_ids);
+        self.user_events.extend(new_results.user_events);
+    }
+
+    pub fn update_version_and_previous_tx(
+        &mut self,
+        lamport_version: SequenceNumber,
+        prev_tx: TransactionDigest,
+    ) {
+        for (id, mut obj) in self.written_objects.iter_mut() {
+            // TODO: All of the following is no longer necessary and can be simplified.
```

Comment:
if this todo is still applicable can you make it more clear what the simplification would be?